### PR TITLE
Framework: update wpcom libs

### PR DESCRIPTION
**Pull request by retrofox**
This PR updates:

* `wpcom.js` to `5.2.0`
* `wpcom-proxy-request` to `3.0.0`
* `wpcom-xhr-request` to `1.0.0`.

The main changes in this release will affect the following behaviors in calypso:

* Catch error for `WP-API` requests in `envelope` mode. Related with (#5682)
* `_headers` field isn't into the body response anymore. Instead of this now the response has a third `headers` parameter.
* Updated `wpcom-xhr-wrapper` to these new changes (b8eeefe0a9c7d9584f54727d2bbd6b9b8149455e)

### wpcom-xhr-wrapper

I've removed almost the whole code of this wrapper. I only left the part that checks the auth in the request which also slightly changed.
We have a layer to create a [WPError]( https://github.com/Automattic/wp-error) so now we can check for an error using `name` property instead of `error`. Specifically for this wrapper:

```es6
if ( error && error.name === 'InvalidTokenError' ) {
  // ...
};
```

I've added [some tests](https://github.com/Automattic/wpcom-xhr-request/blob/master/test/index.js#L240-L262) in `wpcom-xhr-request` to check this instance.

### Testing

Many files have changed in this PR:

client/lib/cart/store/cart-synchronizer.js                             
client/lib/countries-list/README.md
client/lib/countries-list/index.js
client/lib/features-list/index.js
client/lib/notification-settings-store/index.js
client/lib/people/test/fixtures/actions.js
client/lib/plans-list/index.js
client/lib/products-list/index.js
client/lib/reader-site-store/index.js
client/lib/reader-site-store/test/index.js
client/lib/states-list/index.js
client/lib/user-settings/index.js
client/lib/wp/sync-handler/index.js
client/state/domains/suggestions/actions.js
client/state/jetpack-connect/test/actions.js
client/state/jetpack-sync/actions.js
client/state/jetpack-sync/test/actions.js
client/state/jetpack-sync/test/reducer.js
client/state/plans/actions.js
client/state/plans/test/actions.js
client/state/plans/test/fixture.js
client/state/plans/test/reducer.js
client/state/posts/actions.js
client/state/products-list/actions.js
client/state/purchases/actions.js
client/state/purchases/test/actions.js
client/state/reader/feeds/actions.js
client/state/reader/sites/actions.js
client/state/reader/sites/reducer.js
client/state/reader/start/test/actions.js
client/state/sites/media-storage/test/actions.js
client/state/sites/plans/actions.js
client/state/sites/test/actions.js
client/state/stats/lists/actions.js
client/state/stored-cards/actions.js
client/state/terms/actions.js                                                                    

So we should try to tests as much as possible the whole application since we are making requests all the time.
Also we should pay special attention to the changes done in `wpcom-xhr-wrapper` trying to check the right error handling and the authorization check. cc @johngodley.
Everything should be ok.

Test live: https://calypso.live/?branch=update/wpcom-proxy-request

### CODE REVIEW

File: client/lib/user-settings/index.js [hunk 1]
```diff
@@ -122,10 +122,11 @@ UserSettings.prototype.saveSettings = function( callback, settingsOverride ) {
 			} else {
 				// Removed freshly saved data from unsavedSettings
 				keys( data )
-					.filter( x => x !== '_headers' )
 					.forEach( x => delete this.unsavedSettings[ x ] );
 			}
 
+			console.log( 'data: %o', data );
```
Comment (gwwar) [inline client/lib/user-settings/index.js]: left in a console.log

**Comment by gwwar** [comment]
I think things are working normally, but this is a pretty large changeset, so we should probably run through the e2e tests locally to make sure main flows work. https://github.com/Automattic/wp-e2e-tests (Note that this branch likely needs a rebase, due to changes in the plan workflow)

**Comment by lamosty** [comment]
I tried to review this PR but found it quite challenging and time-consuming. This is partly because I don't have much experience in these areas so far but also because the changes are too big.

I think moving the code to ES6 and fixing eslint warnings produces the most noise here. Without them, it would be far smaller. So, I recommend that:

1. Either we move the code to ES6 and fix eslint warnings in different PR;
2. Or commit the "real" changes in one commit and then the fix for eslint warnings + ES6 code in the second commit with proper commit title (like "fix warnings and move to ES6"). That way, we know that here you changed the code and there you just fixed the warnings/ES6 code.

Personally, I would go with the second option. But anyways, just wanted to drop my 2cents. :)

Code is working for me.

**Comment by gwwar** [comment]
Once this is rebased, @hoverduck do you mind running the e2e tests locally against this branch? It'll be easy to miss something with just manual testing.

**Comment by hoverduck** [comment]
@gwwar Sure thing, just ping me here or on Slack when it's ready and I'll kick them off.

File: client/lib/user-settings/index.js (hunk 1 above)
Comment (retrofox) [inline client/lib/user-settings/index.js]: :-o thanks!

File: package.json [hunk 2]
```diff
@@ -113,7 +113,7 @@
     "srcset": "1.0.0",
     "store": "1.3.16",
     "striptags": "2.1.1",
-    "superagent": "1.2.0",
+    "superagent": "^2.1.0",
```
Comment (retrofox) [inline package.json]: ok

**Comment by retrofox** [comment]
> Either we move the code to ES6 and fix eslint warnings in different PR;
> Or commit the "real" changes in one commit and then the fix for eslint warnings + ES6 code in the second commit with proper commit title (like "fix warnings and move to ES6"). That way, we know that here you changed the code and there you just fixed the warnings/ES6 code.
> Personally, I would go with the second option. But anyways, just wanted to drop my 2cents. :)

> Code is working for me.

Yes, I think you're right. I'll try to split and re-organize the commits here. Thanks @lamosty

**Comment by retrofox** [comment]
> @gwwar Sure thing, just ping me here or on Slack when it's ready and I'll kick them off.

Hi @hoverduck. Beyond of some minor changes that we should to do the branch is ready be to tested. Thanks Scott.

**Comment by hoverduck** [comment]
I ran the tests and they were all successful with the exception of signup at the mobile screenwidth.  The Theme Preview page is failing to load.  It works fine when you're logged in (or at tablet/desktop width), so I'm guessing there's some component or data that isn't loaded yet.  I don't see any errors in the console logs.

Mobile:
wp-calypso-6947-mobile

Desktop:
wp-calypso-6947-desktop

Side note - the theme preview is a new step in the signup chain that's not present in production right now.

**Comment by gwwar** [comment]
@retrofox if you fix up the note I left about package.json and `make shrinkwrap` I think this is good to 🚢 We should wait until morning your time to deploy though.

**Comment by gwwar** [comment]
@retrofox what's the status on this one?

**Comment by retrofox** [comment]
Sorry I'll update and rebase in few hours.

**Comment by retrofox** [comment]
@gwwar updated now :-D

**Comment by gwwar** [comment]
Sorry, could you rebase again with master, and generate the shinkwrap again? Let's try to 🚢  this tomorrow

**Comment by retrofox** [comment]
@gwwar let's 🚢 tomorrow

**Pull request merged**